### PR TITLE
Addapache

### PR DESCRIPTION
**Pull request by BenediktMKuehne**
This adds the Apache Server as wsgi-server to the frontend.

Some things to think about:

- What would you say is the max file-size a firmwares are allowed to have?(for upload)

- Currently the -D (dev) and the default installation are compatible with each other. keep or change?

### CODE REVIEW

**Comment by m-1-k-3** [comment]
* What would you say is the max file-size a firmwares are allowed to have?(for upload)

I think it should be defaulted to 250MB max and adjustable easily via configuration.

* Currently the -D (dev) and the default installation are compatible with each other. keep or change?

Is there a reason to not keep?

**Comment by BenediktMKuehne** [comment]
> * Currently the -D (dev) and the default installation are compatible with each other. keep or change?
> 
> Is there a reason to not keep?

For debugging of the wsgi and the apache configs(ssl etc) they would have to become incompatible.
But personally I would keep the dev installation as-is.

**Comment by p4cx** [comment]
> I think it should be defaulted to 250MB max and adjustable easily via configuration.

I especially like this idea. I think we should add all configurations (which are relevant, after EMBArk is running) to a single config file and then the administrator acc can set these in his settings inside of the webui.

**Comment by BenediktMKuehne** [comment]
**Short description of what I've done in this PR:**

- changed some locations of files
- overhauled the installation process (-d is default , -F is Dev etc)
- Added Apache standalone with user-acc and its own directory
- split settings for deploy and development
- enabled docker-linting (except Dockerfile)
- changes that will be needed for docker-deployment
- uninstall option (not 100% yet)
- fixed some csrf stuff

**Comment by BenediktMKuehne** [comment]
`-F` option works now as expected

File: README.md [hunk 1]
```diff
@@ -32,9 +32,12 @@ Furthermore *EMBArk* improves the data provision by aggregating the various scan
 *EMBArk* is using the firmware security scanner *EMBA* as backend. You can take a look at the [*EMBA*](https://github.com/e-m-b-a/emba) repository and get more [information about usage of *EMBA* in the wiki](https://github.com/e-m-b-a/emba/wiki/Usage). Additionally you should check the [*EMBArk* wiki](https://github.com/e-m-b-a/embark/wiki).
 
 ## Start Server
-To start the EMBArk-Server simply run `$ sudo ./embark/run-server.sh`.
+To start the EMBArk-Server simply run `$ sudo pipenv run ./run-server.sh`.
```
Comment (m-1-k-3) [inline README.md]: in the installation script you are refering to
`Ready to use $sudo ./run-server.sh `

File: README.md [hunk 2]
```diff
@@ -32,9 +32,12 @@ Furthermore *EMBArk* improves the data provision by aggregating the various scan
...
 This starts the http-Server on 0.0.0.0:80
 
+## Developer
+For devs we recommend simply using: `sudo ./installer.sh -D ` and the `./dev-tools/debug-server-start.sh` script.
```
Comment (m-1-k-3) [inline README.md]: For developers ...

File: installer.sh [hunk 3]
```diff
@@ -16,35 +16,29 @@
...
-  # echo -e "$CYAN-r$NC         Reinstallation of EMBArk with all dependencies (cleanup of docker environment first)"
-  # echo -e "$RED               ! This deletes all Docker-Images as well !""$NC"
-  echo -e "$CYAN-e$NC         Install EMBA only"
   echo -e "$CYAN-h$NC         Print this help message"
-  echo -e "$CYAN-d$NC         Build EMBArk"  # Webserver on host
-  echo
+  echo -e "$CYAN-d$NC         EMBArk default installation"
+  echo -e "$CYAN-F$NC         Installation of EMBArk for developers"
+  echo -e "$CYAN-e$NC         Install EMBA only"
+  echo -e "$CYAN-D$NC         Install for Docker deployment"
```
Comment (m-1-k-3) [inline installer.sh:32]: Developer installation mode

File: installer.sh [hunk 4]
```diff
@@ -54,28 +48,33 @@ install_emba() {
...
-  #   echo -e "$GREEN""$BOLD""Remove failed docker image""$NC"
-  #   docker image rm "$IMAGE_ID"
-  # done
+  docker container stop embark_db
+  docker container stop embark_redis
+  docker container stop embark_server
+  docker container prune -f --filter "label=flag"
+
+
+  #while docker images | grep -qE "\<none\>"; do
```
Comment (m-1-k-3) [inline installer.sh]: if not needed please remove

**Comment by m-1-k-3** [comment]
please finish it up with some final changes ... Currently some tests are running in my environment. If everything runs smooth we can merge afterwards.

Good job!

File: installer.sh (hunk 3 above)
Comment (BenediktMKuehne) [reply installer.sh:32]: changed to `-F`, same as EMBA

Comment (BenediktMKuehne) [reply installer.sh:32]: `-D` i planed for docker.
which isn't ready yet

**Comment by m-1-k-3** [comment]
real a huge step in the right direction. Thank you

**Review by m-1-k-3 — APPROVED** [review]

**Pull request merged**